src/storage/local.rs

```diff
@@ -47,10 +47,13 @@ impl Backend for LocalBackend {
         _version: Option<u32>,
         _loc: Option<&str>,
     ) -> LalResult<Component> {
+        let tar_path = format!("global/{}/0/{}.tar.gz", name, name);
+        let full_tar_path = config_dir().join(tar_path);
+
         Ok(Component {
                 name: name.into(),
                 version: 0,
-                tarball: "".into()
+                tarball: full_tar_path.into_os_string().into_string().unwrap()
```

Comment:
you don't need unwrap here: https://doc.rust-lang.org/std/path/struct.Path.html#method.display